R/bart.R

```diff
@@ -0,0 +1,326 @@
+#' assemble the ensemble.
+#'
+#' There are different ways to fit this model. See the engine-specific pages
+#' for more details:
+#'
+#' \Sexpr[stage=render,results=rd]{parsnip:::make_engine_list("bart")}
+#'
+#' More information on how \pkg{parsnip} is used for modeling is at
+#' \url{https://www.tidymodels.org/}.
+#'
+#' @inheritParams boost_tree
+#' @param prior_terminal_node_coef A coefficient for the prior probability that
+#' a node is a terminal node. In the expression `prior = a * (1 + d)^(-b)`, this
+#' parameter is the `a` coefficient. Values are usually between 0 and one with
+#' a default of 0.95. This effects the baseline probability; smaller numbers
+#' makes the probabilities larger overall.
+#' @param prior_terminal_node_expo An exponent in the prior probability that
+#' a node is a terminal node. In the expression `prior = a * (1 + d)^(-b)`, this
+#' parameter is the `b` coefficient. Values are usually non-negative with
+#' a default of 2 This effects the rate that the prior probability decreases as
+#' the depth of the tree increases. Larger values make deeper trees more
+#' unlikely.
+#' @param prior_outcome_range A positive value that defines the width of a prior
+#' that the predicted outcome is within a certain range. For regression it is
+#' related to the observed range of the data; the prior is the number of standard
+#' deviations of a Gaussian distribution defined by the observed range of the
+#' data. For classification, it is defined as the range of +/-3 (assumed to be
+#' on the logit scale). The default value is 2.
+#'
+#' @template spec-details
+#'
+#' @template spec-references
+#'
+#' @seealso \Sexpr[stage=render,results=rd]{parsnip:::make_seealso_list("bart")}
+#'
+#' @examples
+#' show_engines("bart")
+#'
+#' bart(mode = "regression", trees = 5)
+#'
+#' # ------------------------------------------------------------------------------
+#' # Examples for terminal node prior
+#'
+#' library(ggplot2)
+#' library(dplyr)
+#'
+#' prior_test <- function(coef = 0.95, expo = 2, depths = 1:10) {
+#'   tidyr::crossing(coef = coef, expo = expo, depth = depths)  %>%
+#'     mutate(
+#'       `terminial node prior` =  coef * (1 + depth)^(-expo),
+#'       coef= format(coef),
+#'       expo = format(expo))
+#' }
+#'
+#' prior_test(coef = c(0.05, 0.5, .95), expo = c(1/2, 1, 2)) %>%
+#'   ggplot(aes(depth, `terminial node prior`, col = coef)) +
+#'   geom_line() +
+#'   geom_point() +
+#'   facet_wrap(~ expo)
+
+#' @export
+bart <-
+  function(mode = "unknown", engine = "dbarts",
+           trees = NULL, prior_terminal_node_coef = NULL,
+           prior_terminal_node_expo = NULL,
+           prior_outcome_range = NULL) {
+
+    args <- list(
+      trees    = enquo(trees),
+      prior_terminal_node_coef  = enquo(prior_terminal_node_coef),
+      prior_terminal_node_expo = enquo(prior_outcome_range),
+      prior_outcome_range = enquo(prior_terminal_node_expo)
```

Comment:
This should be swapped right?

```suggestion
      prior_terminal_node_expo = enquo(prior_terminal_node_expo),
      prior_outcome_range = enquo(prior_outcome_range)
```